src/object/cli_obj.c

```diff
@@ -1532,6 +1652,15 @@ dc_obj_update(tse_task_t *task)
 		goto out_task;
 	}
 
+	rc = obj_rw_bulk_prep(obj, args->iods, args->sgls, args->nr, true,
+			      obj_auxi->fw_shard_tgts != NULL, task,
+			      obj_auxi);
+	if (rc != 0) {
+		D_ERROR("update "DF_OID", obj_rw_bulk_prep failed %d.\n",
+			DP_OID(obj->cob_md.omd_id), rc);
+		goto out_task;
+	}
```

Comment:
Does the bulk space will be re-used for re-try case? The former allocated bulk maybe leaked?

Reply:
Ah, yes, should move this section code after retry flag checking, thanks.